tests/test_profile.py

```diff
@@ -77,6 +77,30 @@ def test_BasicDatasetProfiler():
     assert expected_expectations.issubset(added_expectations)
 
 
+def test_BasicDatasetProfiler_empty_column():
+    """
+    The profiler should determine that empty columns are of null cardinality and of null type and
+    not to generate expectations specific to types and cardinality categories.
+
+    We verify this by running the basic profiler on a Pandas dataset with an empty column
+    and asserting the number of successful results for the empty columns.
+    """
+    toy_dataset = PandasDataset({"x": [1, 2, 3], "y": [None, None, None]}, data_asset_name="toy_dataset")
+    assert len(toy_dataset.get_expectation_suite(
+        suppress_warnings=True)["expectations"]) == 0
+
+    expectations_config, evr_config = BasicDatasetProfiler.profile(toy_dataset)
+
+    assert len([result for result in evr_config['results'] if
```

Comment:
This test seems odd -- would it make more sense to transform to a dict and test for the specific expectations we're expecting here, since it seems like the underlying behavior to check is that we make "more" expectastions about `x` than `y`?

Also: given the underlying case that led to this test, let's ensure we add columns with:
1) non-numeric values (high and low cardinality)
2) high cardinality numeric
3) partially-null

Such example datasets already exist in the test suite -- will help grab them.